extras/create_snapshots_reports_scorecard.py

```diff
@@ -215,49 +225,147 @@ def create_weekly_snapshots(db, cyhy_db_section):
             )
         ]
     )
-    # request_list = ['49ers', 'USAID', 'Hawaii', 'DAS-BEST', 'COC', 'DHS', 'COLA', 'COA', 'COGG', 'AGRS', 'FEC', 'FHFA', 'FMC', 'LPG', 'MSFG', 'OGE', 'PRC', 'SJI', 'USAB', 'VB']
-    # request_list = ['49ers', 'USAID', 'Hawaii', 'DAS-BEST', 'COC', 'DHS', 'COLA', 'COA', 'COGG']
-
-    for i in request_list:
-        # If the customer is in the decsendant org list then don't snap, add to successful
-        # Assume that parent orgs will always be snapped first
-        if i in successful_descendant_snaps:
-            successful_snaps.append(i)
-            logging.info("Added to successful snaps (descendant_snapshot): %s", i)
-            continue
-
-        snap_time = time.time()
-        output = subprocess.Popen(
-            ["cyhy-snapshot", "--section", cyhy_db_section, "create", i],
-            stdout=subprocess.PIPE,
-            stdin=subprocess.PIPE,
-            stderr=subprocess.PIPE,
+
+
+def create_list_of_snapshots_to_generate(db, reports_to_generate):
+    """Create list of organizations that need snapshots generated."""
+    # Find all descendants of orgs that get reports and have children
+    report_org_descendants = set()
+    for i in db.RequestDoc.collection.find(
+        {
+            "report_period": REPORT_PERIOD.WEEKLY,
+            "report_types": REPORT_TYPE.CYHY,
+            "children": {"$exists": True, "$ne": []},
+        },
+        {"_id": 1},
+    ):
+        report_org_descendants.update(db.RequestDoc.get_all_descendants(i["_id"]))
+
+    # Create the list of snapshots to generate by removing
+    # report_org_descendants (their snapshots will be created when their
+    # parent org's snapshot is created)
+    return sorted(list(set(reports_to_generate) - report_org_descendants))
+
+
+def make_list_chunks(my_list, num_chunks):
+    """Split a list into a specified number of smaller lists."""
+    for i in range(0, num_chunks):
+        yield my_list[i::num_chunks]
+
+
+def create_snapshot(db, cyhy_db_section, org_id, use_only_existing_snapshots):
+    """Create a snapshot for a specified organization."""
+    snapshot_start_time = time.time()
+
+    snapshot_command = ["cyhy-snapshot", "--section", cyhy_db_section, "create"]
+
+    if use_only_existing_snapshots:
+        snapshot_command.extend(["--use-only-existing-snapshots", org_id])
+    else:
+        snapshot_command.append(org_id)
```

Comment:
Just a thought to reduce tracking a reference twice.

```suggestion
    if use_only_existing_snapshots:
        snapshot_command.append("--use-only-existing-snapshots")

    snapshot_command.append(org_id)
```

Reply:
Good catch @mcdonnnj - thanks! 👍   
See https://github.com/cisagov/cyhy-reports/pull/61/commits/4024191d65651384dc1038bb4f97f41b775c62f4.